semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/postgres/PostgreSQLVectorStoreQueryProvider.java

```diff
@@ -0,0 +1,302 @@
+            + getColumnNamesAndTypes(dataDeclaredFields, supportedDataTypes) + ", "
+            + getColumnNamesAndTypesForVectorFields(recordDefinition.getVectorFields(), recordClass)
+            + ");";
+
+        String insertCollectionQuery = "INSERT INTO " + validateSQLidentifier(collectionsTable)
+            + " (collectionId) VALUES (?)";
+
+        try (Connection connection = dataSource.getConnection();
+            PreparedStatement createTable = connection.prepareStatement(createStorageTable)) {
+            createTable.execute();
+        } catch (SQLException e) {
+            throw new SKException("Failed to create collection", e);
+        }
+
+        try (Connection connection = dataSource.getConnection();
+            PreparedStatement insert = connection.prepareStatement(insertCollectionQuery)) {
+            insert.setObject(1, collectionName);
+            insert.execute();
+        } catch (SQLException e) {
+            throw new SKException("Failed to insert collection", e);
+        }
+    }
+
+    private void setStatementValues(PreparedStatement statement, Object record,
+        List<VectorStoreRecordField> fields) {
+        for (int i = 0; i < fields.size(); ++i) {
+            VectorStoreRecordField field = fields.get(i);
+            try {
+                Field recordField = record.getClass().getDeclaredField(field.getName());
+                recordField.setAccessible(true);
+                Object value = recordField.get(record);
+
+                if (field instanceof VectorStoreRecordKeyField) {
+                    statement.setObject(i + 1, (String) value);
+                } else if (field instanceof VectorStoreRecordVectorField) {
+                    Class<?> vectorType = record.getClass().getDeclaredField(field.getName())
+                        .getType();
+
+                    // If the vector field is other than String, serialize it to JSON
+                    if (vectorType.equals(String.class)) {
+                        statement.setObject(i + 1, value);
+                    } else {
+                        // Serialize the vector to JSON
+                        statement.setString(i + 1, new ObjectMapper().writeValueAsString(value));
+                    }
+                } else {
+                    statement.setObject(i + 1, value);
+                }
+            } catch (NoSuchFieldException | IllegalAccessException | SQLException e) {
+                throw new SKException("Failed to set statement values", e);
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private String getWildcardStringWithCast(List<VectorStoreRecordField> fields) {
+        StringBuilder wildcardString = new StringBuilder();
+        int wildcards = fields.size();
+        for (int i = 0; i < wildcards; ++i) {
+            if (i > 0) {
+                wildcardString.append(", ");
```

Comment:
When I see this pattern, I can't help but wonder if this loop can be done with a stream and Collectors.joining(", ")

Reply:
I added that suggestion to the parent class, in a similar method. This one has an if statement afterwards, so indexes are needed anyways.